packages/next-tinacms-dos/src/handlers.ts

```diff
@@ -115,19 +123,46 @@ async function uploadMedia(
       id: prefix + filename,
       filename,
       directory: prefix,
-      previewSrc: cdnUrl + prefix + filename,
-      src: cdnUrl + prefix + filename,
+      previewSrc:
+        cdnUrl +
+        (mediaRoot
+          ? path.join(mediaRoot, prefix + filename)
+          : prefix + filename),
+      src:
+        cdnUrl +
+        (mediaRoot
+          ? path.join(mediaRoot, prefix + filename)
+          : prefix + filename),
     })
   } catch (e) {
     res.status(500).send(findErrorMessage(e))
   }
 }
 
+function stripMediaRoot(mediaRoot: string, key: string) {
+  if (!mediaRoot) {
+    return key
+  }
+  const mediaRootParts = mediaRoot.split('/').filter((part) => part)
+  if (!mediaRootParts || !mediaRootParts[0]) {
+    return key
+  }
+  const keyParts = key.split('/').filter((part) => part)
+  // remove each part of the key that matches the mediaRoot parts
+  for (let i = 0; i < mediaRootParts.length; i++) {
+    if (keyParts[0] === mediaRootParts[i]) {
```

Comment:
Wondering is this would cause an issue if you had two nested directories with the same name:

```
mediaRoot: 'blah/nested/blah'
```

Reply:
I don't think so - just tested it and it seems fine. Did find another issue while testing it, though, and fixed that